processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -74,6 +77,76 @@
 {
   private volatile DateTime maxIngestedEventTime;
 
+  private static final Map<String, ValueType> TYPE_MAP = ImmutableMap.<String, ValueType>builder()
+      .put("Long[]", ValueType.LONG)
```

Comment:
then why there is a need for `[]` types here as `TYPE_MAP` is used like `TYPE_MAP.get(singleVal.getClass().getSimpleName())` where singleValue as you said "would be single numerics or Strings for the JSON parsers and all single strings for the delimited parsers, I believe."